crates/store/re_log_encoding/src/decoder/mod.rs

```diff
@@ -153,7 +180,43 @@ impl<R: std::io::Read> Decoder<R> {
         Ok(Self {
             version,
             compression,
-            read,
+            read: Reader::Raw(read),
+            uncompressed: vec![],
+            compressed: vec![],
+        })
+    }
+
+    /// Instantiates a new multiplexed decoder.
+    ///
+    /// This will gracefully handle multiplexed RRD streams, at the cost of extra performance
+    /// overhead, by looking ahead for potential `FileHeader`s in the stream.
+    ///
+    /// The [`CrateVersion`] of the final, demultiplexed stream will correspond to the most recent
+    /// version among all the versions found in the stream.
+    ///
+    /// This is particularly useful when working with stdio streams.
+    ///
+    /// If you're not familiar with multiplexed RRD streams, then you probably want to use
+    /// [`Decoder::new`] instead.
```

Comment:
Why do we need both constructors though? They look identical, except one requires a buffered reader.

Isn't it just better if we have one constructor that always handled concatenated streams?

Reply:
Three reasons:
1. As mentioned above, buffering is a net loss is some real world cases that we depend on today.
2. There's non negligible overhead to constantly check for unexpected `FileHeader`s, which there's really no reason to be paying for in most cases.
3. This is a very specific constructor that relies specifically on `std::io::BufReader`, as opposed to any type that implements `std::io::BufRead`.